File: Python/Majority Element.py
class Solution:
    # @param num, a list of integers
    # @return an integer
    def majorityElement(self, num):
        result = num[0]
        times = 1
        for i in range(1, len(num)):
            if times == 0:
                result = num[i]
                times = 1
            elif num[i] == result:
                times += 1
            elif num[i] != result:
                times -= 1
        return result

File: C++/Majority-Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int res = nums[0], cur = 1;
        for (int i = 1; i < nums.size(); i ++) {
        	if (nums[i] != res) {
        		cur --;
        		if (cur == 0) {
        			res = nums[i];
        			cur = 1;
        		}
        	} else {
        		cur ++;
        	}
        }
        return res;
    }
};